Remove direct pricing info and disabled regions (#16901)
cockroachdb/docs: _includes/cockroachcloud/cockroachcloud-pricing.md

> Pricing depends on your cloud provider and region selections. See the [Pricing](https://www.cockroachlabs.com/pricing/) page for more information.

Reviewer: ```suggestion
Pricing depends on your cloud provider and region selections. Refer to [Pricing](https://www.cockroachlabs.com/pricing/) for details.
```
